app_legacy/Helpers/database/player-achievement.php

```diff
@@ -150,6 +150,8 @@ function unlockAchievement(string $user, int $achIDToAward, bool $isHardcore): a
     $retVal['Success'] = true;
     // Achievements all awarded. Now housekeeping (no error handling?)
 
+    expireUserAchievementUnlocksForGame($user, $achData['GameID']);
```

Comment:
This causes the `getUserAchievementUnlocksForGame` cache entry to be discarded when an achievement is unlocked.